nova/virt/vmwareapi/vm_util.py

```diff
@@ -1738,11 +1739,25 @@ def _process_host_stats(obj, host_reservations_map):
 
     _set_hypervisor_type_and_version(stats, host_props)
     _set_host_reservations(stats, host_reservations_map, obj.obj)
-    return host_props["name"], stats
+    return stats
 
 
-def get_stats_from_cluster_per_host(session, cluster):
-    """Get the resource stats per host of a cluster."""
+def get_hosts_with_stats(session, cluster):
+    """Get the hosts with the underlying resource stats.
+
+    Returns a dict containing the host mo-ref as key, and the
```

Comment:
Not the mo-ref but only the mo-ref value. We need to be precise, because one is a string and the other an object - the object is different every time we fetch it from VC.